ui/perfherder/perf-helpers/helpers.js

```diff
@@ -867,3 +868,75 @@ export const getResultsMap = (results) => {
   });
   return { testNames, names, platforms, resultsMap };
 };
+
+export const getTimeDifference = (startDate, endDate) => {
+  const timeDifference = Math.abs(endDate - startDate);
+  let daysDifference = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
+
+  // If weekday of startDate is bigger than endDate then it is a new week,
+  // and we have to subtract the weekend days
+  // [1. Mon] [2. Tue] [3. Wed] [4. Thu]  [5. Fri] [6. Sat] [7. Sun]
+  //                            startDate -------- -------- ----->>>
+  // -------- -------> endDate
+  if (startDate.getUTCDay() > endDate.getUTCDay()) {
+    daysDifference -= 2;
+  } else {
+    daysDifference -= 2 * Math.trunc(daysDifference / 7);
+  }
+
+  let hoursDifference = Math.ceil(timeDifference / (1000 * 60 * 60));
+  // If due date is Monday and today is Friday and we have to show the hours left,
+  // we have to subtract the weekend hours from the difference
+  if (
+    startDate.getUTCDay() === weekdays.friday &&
+    endDate.getUTCDay() === weekdays.monday
+  ) {
```

Comment:
```suggestion
const shouldGetHoursLeft = startDate.getUTCDay() === weekdays.friday && endDate.getUTCDay() === weekdays.monday
  if ( shouldGetHoursLeft ) {
```
I am open to other suggestions for naming the constant.